gdrdrv/gdrdrv.c

```diff
@@ -479,16 +493,23 @@ static int gdrdrv_pin_buffer(gdr_info_t *info, void __user *_params)
         }
     }
 
-
     // here a typical driver would use the page_table to fill in some HW
     // DMA data structure
 
-    params.handle = mr->handle;
-
     mutex_lock(&info->lock);
     list_add(&mr->node, &info->mr_list);
+
+    // The user treats handle as the offset value for mmap. To guarantee no
```

Comment:
What about something like:
// the user-space library passes the memory handle as the mmap offset,
// and offsets are used to determine the VMAs to delete during invalidation.
// Hence, we need [handle,handle+size-1] to correspond to a unique VMA.
// Note that size here must match the original mmap size